homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -62,6 +146,114 @@ def update(self, new_data: PassiveBluetoothDataUpdate[_T]) -> None:
         self.entity_data.update(new_data.entity_data)
         self.entity_names.update(new_data.entity_names)
 
+    def async_get_restore_data(self) -> RestoredPassiveBluetoothDataUpdate:
+        """Serialize restore data to storage."""
+        return {
+            "devices": {
+                key or "": device_info for key, device_info in self.devices.items()
```

Comment:
should this also be `key.to_string()` ?

Reply:
I guess `set_restore_data` assumes it's not .

Reply:
The device ids are either `str` or `None` but since we can't serialize `None` as a JSON key  because the spec doesn't allow non-strings so it has to be `""`.  The restore goes the other way